### PR TITLE
Remove tween.js and use a home-rolled animator instead

### PR DESCRIPTION
**Pull request by blowery**
while this is fancy, do we want to simply move to a direct animation without the scroll? would save all this code.

### CODE REVIEW

**Comment by blowery** [comment]
Looking at test failures.

File: client/lib/scroll-to/index.js [hunk 1]
```diff
@@ -17,34 +12,112 @@ function getCurrentScroll( container ) {
...
+	};
+}
+
+function circularInOutEasing( val ) {
+	if ( ( val *= 2 ) < 1 ) {
+		return -0.5 * ( Math.sqrt( 1 - val * val ) - 1 );
 	}
 
-	TWEEN.update();
+	return 0.5 * ( Math.sqrt( 1 - ( val -= 2 ) * val ) + 1 );
```
Comment (jsnajdr) [inline client/lib/scroll-to/index.js]: Where does this formula come from and what shape it is supposed to produce? The way how `val` is mutated in mid-expression makes is harder to understand than neccessary.

Comment (blowery) [reply client/lib/scroll-to/index.js]: It's the circular inout easing from tween.js

**Comment by blowery** [comment]
> Element.scrollIntoView has a behavior: 'smooth' option. Would that work? 

Nope, it doesn't take fixed overlays, like the masterbar, into account. We add a custom offset to the destination to make sure it's actually on screen.

**Comment by samouri** [comment]
This has a nice effect on the build chunk: 

```
Commit: b7d51514d7dd291ac48ee0b428ae2153931eac96
Author: blowery
At: 2018-03-27T20:44:03Z
Message: Remove tween.js and use a home-rolled animator instead
Delta:
chunk              stat_size           parsed_size           gzip_size
async-load-design      +38 B  (+0.0%)        +60 B  (+0.0%)      +12 B  (+0.0%)
build               -10758 B  (-0.3%)      -4437 B  (-0.3%)    -1475 B  (-0.4%)
checkout               +38 B  (+0.0%)        +60 B  (+0.0%)       +6 B  (+0.0%)
manifest                +0 B                  +0 B                -2 B  (-0.1%)
plugins                +38 B  (+0.0%)        +60 B  (+0.0%)       +4 B  (+0.0%)
vendor                +132 B  (+0.0%)        +38 B  (+0.0%)      +13 B  (+0.0%)
woocommerce            +38 B  (+0.0%)        +60 B  (+0.0%)       +9 B  (+0.0%)
```

**Comment by samouri** [comment]
This definitely works for the basic happy case, I tested with keyboard navigation in Reader.
Are there any edge cases we should look out for?

**Comment by samouri** [comment]
Just tested this out again, same question as before:

> This definitely works for the basic happy case, I tested with keyboard navigation in Reader.
Are there any edge cases we should look out for?

**Comment by blowery** [comment]
> Are there any edge cases we should look out for?

Hitting `j` really quickly / held down. Ideally it's better than with tween, at worst it's the same. 

Should also test other things that use `lib/scroll-to`:
```
client/reader/stream/index.jsx:36:import scrollTo from 'lib/scroll-to';
client/blocks/reader-full-post/index.jsx:38:import scrollTo from 'lib/scroll-to';
client/layout/guided-tours/tours/main-tour.js:30:import scrollTo from 'lib/scroll-to';
client/layout/guided-tours/tours/jetpack-basic-tour.js:30:import scrollTo from 'lib/scroll-to';
client/layout/guided-tours/positioning.js:13:import scrollTo from 'lib/scroll-to';
client/my-sites/comments/comment/comment-actions.jsx:17:import scrollTo from 'lib/scroll-to';
client/my-sites/comments/comment/index.jsx:24:import scrollTo from 'lib/scroll-to';
client/my-sites/domains/components/form/country-select.jsx:21:import scrollIntoViewport from 'lib/scroll-into-viewport';
client/my-sites/domains/components/form/input.jsx:19:import scrollIntoViewport from 'lib/scroll-into-viewport';
client/my-sites/domains/components/form/state-select.jsx:25:import scrollIntoViewport from 'lib/scroll-into-viewport';
client/my-sites/checkout/cart/secondary-cart.jsx:25:import scrollIntoViewport from 'lib/scroll-into-viewport';
client/my-sites/stats/activity-log-item/index.jsx:8:import scrollTo from 'lib/scroll-to';
client/my-sites/stats/activity-log/index.jsx:8:import scrollTo from 'lib/scroll-to';
client/components/infinite-list/index.jsx:23:import scrollTo from 'lib/scroll-to';
client/post-editor/editor-diff-viewer/index.jsx:20:import scrollTo from 'lib/scroll-to';
client/lib/scroll-to/README.md:9:import scrollTo from 'lib/scroll-to';
```

**Comment by samouri** [comment]
I have no idea how to activate all of these, but I figure I triggered the changed codepath with what I did test.

**testing done**
1. scrolls like a literal madman on reader streams ✔️ 
2. launch a reader full post page straight to a comment ✔️ 
3. launched a guided tour and clicked thru ✔️ 

:shipit:

**Review by samouri — APPROVED** [review]

**Pull request merged**